reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscription.scala

```diff
@@ -76,18 +77,17 @@ private[reactivestreams] final class StreamSubscription[F[_], A](
         .compile
         .drain
 
-    dispatcher.unsafeRunAndForget(s)
+    startDispatcher.unsafeRunAndForget(s)
   }
 
   // According to the spec, it's acceptable for a concurrent cancel to not
   // be processed immediately, but if you have synchronous `cancel();
-  // request()`, then the request _must_ be a no op. For this reason, we
-  // need to make sure that `cancel()` does not return until the
-  // `cancelled` signal has been set.
+  // request()`, then the request _must_ be a no op. Fortunately,
+  // ordering is guaranteed by a sequential d
   // See https://github.com/zainab-ali/fs2-reactive-streams/issues/29
   // and https://github.com/zainab-ali/fs2-reactive-streams/issues/46
   def cancel(): Unit =
-    dispatcher.unsafeRunSync(cancelled.set(true))
+    requestDispatcher.unsafeRunAndForget(cancelled.set(true))
```

Comment:
So this is interesting. Previously `unsafeRunSync` was necessary to achieve this behavior. Now the shared sequential `Dispatcher` guarantees the required ordering out-of-the box.